File: C/gnomesort/qwen-qwq-32b.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

void gnomeSort(int arr[], int n) {
    int i = 0;
    while (i < n) {
        if (i == 0) {
            i++;
            continue;
        }
        if (arr[i] >= arr[i-1]) {
            i++;
        } else {
            int temp = arr[i];
            arr[i] = arr[i-1];
            arr[i-1] = temp;
            i--;
        }
    }
}

int main() {
    int arr[1000];
    int size = 0;

    FILE *file = fopen("../../inputs.txt", "r");
    if (!file) {
        return 1;
    }

    while (fscanf(file, "%d", &arr[size]) == 1 && size < 1000) {
        size++;
    }
    fclose(file);

    gnomeSort(arr, size);

    for (int i = 0; i < size; ++i) {
        printf("%d ", arr[i]);
    }
    printf("\n");

    return 0;
}

File: Python/gnomesort/qwen-qwq-32b.py
def gnome_sort(arr):

    i = 1

    n = len(arr)

    while i < n:

        if arr[i] < arr[i-1]:

            arr[i], arr[i-1] = arr[i-1], arr[i]

            i -=1

            if i <1:

                i =1

        else:

            i +=1

    return arr

import sys

def main():
    input_path = "inputs.txt" if len(sys.argv) < 2 else sys.argv[1]

    try:
        with open(input_path, 'r') as f:
            arr = [int(line.strip()) for line in f if line.strip()]
    except FileNotFoundError:
        return

    sorted_arr = gnome_sort(arr)
    
    print(sorted_arr)

if __name__ == "__main__":
    main()